docs/nmt-lib.md

```diff
@@ -0,0 +1,187 @@
+              /            \                    /               \
+            NsH()          NsH()              NsH()             NsH() 
+            /                \                /                   \
+    00 00 5fa0c9c       00 00 52385a0    01 01 71ca46a       03 03 b4a2792    Leaf Nodes
+        |                   |                 |                   |
+      NsH()               NsH()              NsH()               NsH()
+        |                   |                 |                   |
+00 6c6561665f30      00 6c6561665f31    01 6c6561665f32      03 6c6561665f33  Namespaced Data Items 
+
+        0                   1                  2                  3           Leaf Indices
+``` 
+Figure 1.
+
+## Get Root
+
+The `Root()` method calculates the NMT root based on the data that has been added through the use of the `Push` method.
+```go
+func (n *NamespacedMerkleTree) Root() []byte
+```
+For example:
+```go
+// compute the root
+root := tree.Root() 
+```
+In the provided code example, the root would be `00 03 b1c2cc5` (as also illustrated in Figure 1).
+
+The minimum and maximum namespace IDs of the tree root can be obtained through the following methods:
+
+```go
+minNS := nmt.MinNamespace(root, tree.NamespaceSize())
+maxNS := nmt.MaxNamespace(root, tree.NamespaceSize())
+```
+The `minNs` and `maxNs` are equal to `00` and `03` in the supplied example.
+
+## Generate Namespace Proof
+
+The `ProveNamespace` method can be used to generate a namespace proof for a specific namespace ID.
+
+```go
+func (n *NamespacedMerkleTree) ProveNamespace(nID namespace.ID) (Proof, error)
+```
+For example:
+```go
+nID := namespace.ID{0}
+proof, err := tree.ProveNamespace(nID)
+if err != nil {
+  panic("unexpected error")
+}
+```
+
+The returned proof is of the following structure:
+
+```go
+type Proof struct {
+	start int
+	end int
+	nodes [][]byte
+	leafHash []byte
+	isMaxNamespaceIDIgnored bool
+}
+```
+The fields can be interpreted as follows:
+
+`start, end`:  They represent the starting index and the ending index of leaves that match the provided namespace ID `nID`.
+Note that `end` is non-inclusive.
+
+`nodes`: The `nodes` hold the tree nodes necessary for the Merkle range proof of `[start, end)`  ordered according to in-order traversal of the tree.
+`nodes` embodies an ordered list of byte slices, where each byte slice contains an NMT node.
+Nodes have identical size and all follow the [namespaced hash format](#namespaced-hash).
+In the example given earlier, each node is `34` bytes in length and takes the following form:  `minNs<1 byte>||maxNs<1 byte>||h<32 byte>`.
+
+`leafHash`: This field is non-empty only for absence proofs and contains a leaf hash required for such a proof (see [namespace absence proofs](#namespace-absence-proof) section).
+
+`isMaxNamespaceIDIgnored`: If this field is present, then namespace range of the tree nodes are set as explained in the [Ignore Max Namespace](#ignore-max-namespace) section.
```

Comment:
[uber nit] it could be present but false so to clarify:

```suggestion
`isMaxNamespaceIDIgnored`: If this field is true, then namespace range of the tree nodes are set as explained in the [Ignore Max Namespace](#ignore-max-namespace) section.
```

Reply:
Addressed https://github.com/celestiaorg/nmt/pull/120/commits/9d50cafbc6dea994a29bc7a768e23d7c35106761